3 new features

schoettler-software opened the pull request:
We used your tabledata script in a project and added some small features which we thought might come in handy for future users. 

- a pagesize selector to change the number of displayed rows dynamically
- an option to use externally defined comparision functions for sorting columns ( we needed a comparision function which replaced german umlaut with corresponding vowels)
- an option to use an externally defined function to call when pagesize is changed ( we used it to store the value of the pagesize)

These additions are backward-compatible, we added a small example to show how our new options are applied.

```
<html>
<title>Example</title>
<link rel="stylesheet" type=""text/css" href="css/datatable.css"/>
<script src="js/datatable.js"></script>

<table id="exampletable" width="400" border="1">       
	<caption><b class="counter">loading</b>
		<div class="paging">Items per Page: <span class="pagesize"></span></div>
	</caption>
	<thead>
		<tr><th>letter</th><th>number</th><th>stable</th></tr>
	</thead>
	<tbody>
		<tr><td>p</td><td>2</td><td>1</td></tr>
		<tr><td>n</td><td>2.2</td><td>1</td></tr>
		<tr><td>ö</td><td>22</td><td>1</td></tr>
		<tr><td>o</td><td>22.2</td><td>2</td></tr>
		<tr><td>n</td><td>2.2</td><td>2</td></tr>
	</tbody>
</table>
<script>
var pageSize = window.localStorage.getItem("pageSize") || 2;

sample = new DataTable(document.getElementById("exampletable"), {
	sort : '*',
	sortAlgorithm : localSort,
	pageSize : pageSize,
	pageSizeOptions: [1, 2, 5],
	onPageSizeChange: function(newPageSize){ window.localStorage.setItem("pageSize", newPageSize); },
	pageSizeDivSelector : ".pagesize"
});
	
function localSort(item1, item2) {

	if( item1.localeCompare !== undefined ) {
		return comparision = item1.localeCompare(item2);
	} else {
		return item1 > item2 ? 1 : (item1 < item2 ? -1 : 0);
	}
}
</script>
```

Commit:
new features from schoettler-software:
- multicolumn-sort via stable sorting-algorithm
- user-selectable pagesizes
- customizable comparison function for sorting

Commit:
created minified file with: uglifyjs -o datatable.min.js datatable.js

Commit:
Updated documentation to options

Holt59 commented:
Thanks for the PR, I'm glad to see that this plugin is used in concrete applications!

I'll take a look at this PR as soon as I can, but I probably won't have time before next week-end.

SamuelKirschner committed:
replaced all tabs with spaces

SamuelKirschner committed:
removed use-case specific code, added the option selectedRow instead.

SamuelKirschner committed:
updated datatable.min.js

schoettler-software commented:
We forgot about one use-case.
Each row in the datatable was clickable, and led to a page where the data could be edited.
On that page it was important that the clicked row was highlighted.
The one feature that was missing, was starting the paginator on the page with the highlighted row.

The code for that use-case was still inside the datatable.js.
I removed the use-case specific code and added an option called `selectedRow`.

In the example I used the following options to demonstrate the behaviour:
```
	lineFormat: function (id, data) {
		var res = document.createElement('tr');
		res.dataset.id = id;
		if(new URLSearchParams(window.location.search).get('index') == id) {
			res.setAttribute('bgcolor', '#00ff00');
		}
		for (var key in data) {
			if (data.hasOwnProperty(key)) {
				res.innerHTML += '<td>' + data[key] + '</td>';
			}
		}
		res.addEventListener('click', function() {
			window.location.href = '?index=' + id;
		}, false);
		return res;
	},
	selectedRow: function(index, data) {
		return index == new URLSearchParams(window.location.search).get('index')
	},
```
-- Samuel Kirschner
